src/site/stages/build/page-builder/transformers.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * A very specific helper function that expects to receive an
+ * array with one item which is an object with a single `value` property
+ *
+ */
+function arrayCollapse(arr) {
+  return arr.reduce((value, item) => {
+    // eslint-disable-next-line no-param-reassign
+    value = item.value;
+    return value;
+  }, null);
+}
+
+function pageTransform(entity) {
+  const transformed = entity;
+  const {
+    title,
+    changed,
+    fieldIntroText,
+    fieldPageLastBuilt,
+    fieldAlert,
+    fieldDescription,
+    moderationState: [{ value: published }],
+  } = entity;
+  // collapse title
+  // Question: Can we always assume that title is an array of one item, with that item being an object with a `value` key?
+  transformed.title = arrayCollapse(title);
+  transformed.entityBundle = 'page';
+
+  transformed.fieldIntroText = arrayCollapse(fieldIntroText);
+  transformed.changed = new Date(arrayCollapse(changed)).getTime() / 1000;
+  transformed.fieldPageLastBuilt = new Date(
+    arrayCollapse(fieldPageLastBuilt),
+  ).toUTCString();
+
+  transformed.entityPublished = published === 'published';
+  delete transformed.moderationState;
+
+  if (Array.isArray(fieldDescription) && fieldDescription.length === 0) {
+    transformed.fieldDescription = null;
+  }
+
+  if (
+    Array.isArray(fieldAlert) &&
+    fieldAlert.length === 1 &&
+    fieldAlert[0].length === 0
+  ) {
+    transformed.fieldAlert = { entity: null };
+  }
+
+  return entity;
+}
+
+const transformers = {
+  page: pageTransform,
+};
+
+// Recursively replaces all snake_case properties in an object
+// with camelCased properties
+function toCamel(obj) {
```

Comment:
These are really fun functions to write! However, I think we may want to go for [Lodash's `camelCase` method](https://lodash.com/docs/4.17.15#camelCase) here since it does a lot of null checking built-in to prevent runtime errors. What do you think @cvalarida?

Reply:
I'm typically a fan of using code somebody else wrote. :smile: In this case, though, what really matters is performance, so I wrote up [a little benchmark](http://jsben.ch/UYzYp).

Looks like lodash's is pretty quick. I'm in favor of using lodash for this.
![image](https://user-images.githubusercontent.com/12970166/67803229-d4b99700-fa49-11e9-8e27-34ce3433ce41.png)

Reply:
It helps to know what dependencies we are already using :man_facepalming:

Reply:
Going off of the docs and my limited testing, it looks like `camelCase` is meant only for strings, so having the benchmarks pass an object to it may be a bit misleading.  This is what I ended up going with:

```js
_.mapKeys(entity, (v, k) => _.camelCase(k));
```

The [benchmark](http://jsben.ch/OECaa) is a bit different, but lodash still seems ok:

![image](https://user-images.githubusercontent.com/2008881/67814902-9d0b1900-fa62-11e9-961a-84002b167f50.png)